tests/pylint/runpylint.py

```diff
@@ -34,20 +40,86 @@ def __init__(self):
                                 FalsePositive(r"E1101.*: Instance of 'int' has no 'dasd_is_ldl' member"),
                               ]
 
-        # This will solve problems with C python extensions
-        self.loadAllExtensions = True
+    def _files(self):
+        srcdir = os.environ.get("top_srcdir", os.getcwd())
 
-    @property
-    def extraArgs(self):
-        return ["--init-import", "y"]
+        retval = self._get_py_paths(srcdir)
 
-    @property
-    def initHook(self):
-        return """'import gi.overrides, os; gi.overrides.__path__[0:0] = (os.environ["ANACONDA_WIDGETS_OVERRIDES"].split(":") if "ANACONDA_WIDGETS_OVERRIDES" in os.environ else [])'"""
+        return retval
+
+    def _get_py_paths(self, directory):
```

Comment:
Do we really need this scary method?

Reply:
Yes and no. I would say we have a three options here:

* Leave it here and have everything tested.
* Remove it and write every python folder here.
* Remove it and test only pyanaconda and nosetests / pyanaconda tests.

The problem is, that pylint will give you possibility to set a folder and it will find all the python files inside. However, it is great in failing if the folder is not a python module. And we have a plenty of not a python module stuff here :(.

Reply:
I was looking for a better solution, but this one seems to be the most reliable, so let's keep it.